### PR TITLE
Allow special default values for certain columns

### LINKED ISSUE #280: Add handling of special default values
The Builder plugin currently cannot handle certain valid default values for certain datatypes when creating a new table in the **Database** panel. A suitable PR for this issue would adjust the migration generator to allow these certain values in the "Default" column:

- **All fields with nullable ticked**: Should accept `null` as a valid value to be rendered as `null` in the migration. Should also accept `"null"` (with quotes) as a string value of `null` for the (very rare) case that a dev needs null stored as a string in a field.
- **Boolean fields**: Should accept `true` and `false` as valid values to be rendered as `true` and `false` respectively.

### PR DESCRIPTION
**Pull request by bennothommo**
- If column is nullable, allow `null` for a NULL value, and `"null"` as a string "null"
- If column is a boolean, allow `true` and `false` as values.

Fixes #280.

### CODE REVIEW

File: classes/DatabaseTableModel.php [hunk 1]
```diff
@@ -383,6 +384,11 @@ protected function loadColumnsFromTableInfo()
                 'id' => $columnName,
             ];
 
+            // Format quoted "null" values with quotes
+            if ($column->getNotnull() === false && strtolower($item['default'] === 'null')) {
```
Comment (w20k) [inline classes/DatabaseTableModel.php]: @bennothommo maybe you've misplaced the brackets: `strtolower($item['default']) === 'null'`?

File: classes/TableMigrationCodeGenerator.php [hunk 2]
```diff
@@ -490,12 +484,21 @@ protected function generateDefaultMethodCall($default, $column)
...
-        if (in_array($type, MigrationColumnType::getIntegerTypes()) || 
+        if (in_array($type, MigrationColumnType::getIntegerTypes()) ||
             in_array($type, MigrationColumnType::getDecimalTypes()) ||
             $type == MigrationColumnType::TYPE_BOOLEAN) {
             return sprintf('->default(%s)', $default);
         }
 
+        // Allow `null` value for nullable columns
+        if ($column->getNotnull() === false) {
+            if (strtolower($default) === 'null' && $column->getNotnull() === false) {
```
Comment (w20k) [inline classes/TableMigrationCodeGenerator.php]: @bennothommo minor note, could be on one micro-microsecond faster: `$column->getNotnull() === false && strtolower($default) === 'null'` 😄

File: classes/DatabaseTableModel.php (hunk 1 above)
Comment (bennothommo) [reply classes/DatabaseTableModel.php]: @w20k Oops :P

File: classes/TableMigrationCodeGenerator.php (hunk 2 above)
Comment (bennothommo) [reply classes/TableMigrationCodeGenerator.php]: @w20k I just realised the `getNotnull` call is repeated - I've removed that.

**Comment by w20k** [comment]
LGTM @bennothommo!

**Pull request merged**